src/sdk-reference/cpp/1/document/search/index.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: sdk.html.hbs
+algolia: true
+title: search
+description:
+order: 200
+---
+
+# search
```

Comment:
This is taken from the old "search" function documentation. It has been rewritten in the new API documentation, mainly to properly separate arguments and options documentation from the function description itself.

I believe the same model should be used here: https://docs-v2.kuzzle.io/api/1/controller-document/search/
(this is mainly about removing redundancies)